src/unix/sentry_unix_unwinder_libunwindstack.cpp

```diff
@@ -0,0 +1,65 @@
+#include "sentry_unix_unwinder_libunwindstack.h"
+
+#include <memory>
+#include <ucontext.h>
+#include <unwindstack/Elf.h>
+#include <unwindstack/MapInfo.h>
+#include <unwindstack/Maps.h>
+#include <unwindstack/Memory.h>
+#include <unwindstack/Regs.h>
+#include <unwindstack/RegsGetLocal.h>
+
+size_t
+sentry__unwind_stack_unwindstack(
+    void *addr, const sentry_ucontext_t *uctx, void **ptrs, size_t max_frames)
+{
+    std::unique_ptr<unwindstack::Regs> regs;
+
+    if (uctx) {
+        regs = std::unique_ptr<unwindstack::Regs>(
+            unwindstack::Regs::CreateFromUcontext(
+                unwindstack::Regs::CurrentArch(), uctx->user_context));
+    } else if (!addr) {
+        regs = std::unique_ptr<unwindstack::Regs>(
+            unwindstack::Regs::CreateFromLocal());
+        unwindstack::RegsGetLocal(regs.get());
+    } else {
+        return 0;
+    }
+
+    unwindstack::LocalMaps maps;
+    if (!maps.Parse()) {
+        ptrs[0] = (void *)regs->pc();
+        return 1;
+    }
+
+    const std::shared_ptr<unwindstack::Memory> process_memory(
+        new unwindstack::MemoryLocal);
+
+    int rv = 0;
+    for (int i = 0; i < max_frames; i++) {
+        ptrs[rv++] = (void *)regs->pc();
+        unwindstack::MapInfo *const map_info = maps.Find(regs->pc());
+        if (!map_info) {
+            break;
+        }
+
+        // the boolean false parameter disables debugdata loading which we don't
+        // want due to size constraints.  Also that data is unlikely to be
+        // useful anyways.
+        unwindstack::Elf *const elf = map_info->GetElf(process_memory, false);
```

Comment:
These `const` specifiers are sort of strange. Are you sure you don't mean `const Elf *` (equivalent to `Elf const *`)?

Reply:
I was just copy-pasting that code, but also pushed your suggestions.